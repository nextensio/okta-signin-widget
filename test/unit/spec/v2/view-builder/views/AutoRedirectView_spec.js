import { Model } from 'okta';
import AutoRedirectView from 'v2/view-builder/views/AutoRedirectView';
import AppState from 'v2/models/AppState';
import Settings from 'models/Settings';
import SuccessWithAppUser
  from '../../../../../../playground/mocks/data/idp/idx/success-with-app-user.json';
import { INTERSTITIAL_REDIRECT_VIEW } from 'v2/ion/RemediationConstants';


describe('v2/view-builder/views/AutoRedirectView', function() {
  let testContext;
  let settings = new Settings({ 
    baseUrl: 'http://localhost:3000'
  });  
  beforeEach(function() { 
    testContext = {};
    testContext.init = (user = SuccessWithAppUser.user.value, app = SuccessWithAppUser.app.value) => {
      const appState = new AppState();
      appState.set('user', user);
      appState.set('app', app);

      testContext.view = new AutoRedirectView({
        appState,
        settings,
        currentViewState: {},
        model: new Model(),
      });
      testContext.view.render();
    };
  });
  afterEach(function() {
    jest.resetAllMocks();
  });

  it('view renders corectly - default case', function() {
    testContext.init();
    expect(testContext.view.el).toMatchSnapshot();
  });

  it('view renders correctly according to interstitialBeforeLoginRedirect', function() {
    settings = new Settings({ 
      baseUrl: 'http://localhost:3000',
      'interstitialBeforeLoginRedirect': INTERSTITIAL_REDIRECT_VIEW.NONE
    });    
    testContext.init();
    expect(testContext.view.el).toMatchSnapshot('should NOT render spinner');
    

    settings = new Settings({ 
      baseUrl: 'http://localhost:3000',
      'interstitialBeforeLoginRedirect': INTERSTITIAL_REDIRECT_VIEW.DEFAULT
    });    
    testContext.init();
    expect(testContext.view.el).toMatchSnapshot('should have spinner');
  });
});
