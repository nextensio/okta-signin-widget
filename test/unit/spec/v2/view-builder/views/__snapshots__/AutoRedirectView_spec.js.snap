// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v2/view-builder/views/AutoRedirectView view renders corectly - default case 1`] = `
<div
  class="siw-main-view "
>
  <div
    class="siw-main-header"
  >
    <div />
  </div>
  <div
    class="siw-main-body"
  >
    <form
      action="/"
      class="ion-form o-form o-form-edit-mode"
      data-se="o-form"
      id="form1"
      method="POST"
      slot="content"
    >
      <div
        class="o-form-content o-form-theme clearfix"
        data-se="o-form-content"
      >
        <h2
          class="okta-form-title o-form-head"
          data-se="o-form-head"
        >
          Signing in...
        </h2>
        <div
          class="o-form-info-container"
        />
        <div
          class="o-form-error-container"
          data-se="o-form-error-container"
        />
        <div
          class="o-form-fieldset-container"
          data-se="o-form-fieldset-container"
        />
      </div>
    </form>
  </div>
  <div
    class="siw-main-footer"
  >
    <div
      class="auth-footer"
    />
  </div>
</div>
`;

exports[`v2/view-builder/views/AutoRedirectView view renders correctly according to interstitialBeforeLoginRedirect: should NOT render spinner 1`] = `
<div
  class="siw-main-view "
>
  <div
    class="siw-main-header"
  >
    <div />
  </div>
  <div
    class="siw-main-body"
  >
    <form
      action="/"
      class="ion-form o-form o-form-edit-mode"
      data-se="o-form"
      id="form2"
      method="POST"
      slot="content"
    >
      <div
        class="o-form-content o-form-theme clearfix"
        data-se="o-form-content"
      >
        <h2
          class="okta-form-title o-form-head"
          data-se="o-form-head"
        >
          Signing in...
        </h2>
        <div
          class="o-form-info-container"
        />
        <div
          class="o-form-error-container"
          data-se="o-form-error-container"
        />
        <div
          class="o-form-fieldset-container"
          data-se="o-form-fieldset-container"
        />
      </div>
    </form>
  </div>
  <div
    class="siw-main-footer"
  >
    <div
      class="auth-footer"
    />
  </div>
</div>
`;

exports[`v2/view-builder/views/AutoRedirectView view renders correctly according to interstitialBeforeLoginRedirect: should have spinner 1`] = `
<div
  class="siw-main-view "
>
  <div
    class="siw-main-header"
  >
    <div />
  </div>
  <div
    class="siw-main-body"
  >
    <form
      action="/"
      class="ion-form o-form o-form-edit-mode"
      data-se="o-form"
      id="form3"
      method="POST"
      slot="content"
    >
      <div
        class="o-form-content o-form-theme clearfix"
        data-se="o-form-content"
      >
        <h2
          class="okta-form-title o-form-head"
          data-se="o-form-head"
        >
          Signing in to Okta Administration as admin@idx.com
        </h2>
        <div
          class="o-form-info-container"
        />
        <div
          class="o-form-error-container"
          data-se="o-form-error-container"
        />
        <div
          class="o-form-fieldset-container"
          data-se="o-form-fieldset-container"
        >
          <div
            class="okta-waiting-spinner"
          />
        </div>
      </div>
    </form>
  </div>
  <div
    class="siw-main-footer"
  >
    <div
      class="auth-footer"
    />
  </div>
</div>
`;
